src/DataTable/DataTable.svelte

```diff
@@ -144,13 +144,10 @@
   };
   const dispatch = createEventDispatcher();
   const batchSelectedIds = writable(false);
-  const headerItems = writable([]);
   const tableRows = writable(rows);
-  const thKeys = derived(headerItems, () =>
-    headers
-      .map(({ key }) => ({ key, id: key }))
-      .reduce((a, c) => ({ ...a, [c.key]: c.id }), {})
-  );
+  $: thKeys = headers
+    .map(({ key }) => ({ key, id: key }))
```

Comment:
Since we're here, let's refactor this to remove the `.map` since we use the `key` as the `id`.

```js
$: thKeys = headers.reduce((a, c) => ({ ...a, [c.key]: c.key }), {});
```

Reply:
Thanks, it makes sense and works fine in my local testing.